### PR TITLE
test: resolve integration test job arns becoming UUIDs

### PR DESCRIPTION
**Pull request by math411**
*Issue #, if available:*

*Description of changes:*
Fix the issue of ARNs replacing job names in the integ tests. The names assigned for the integ test are unique enough that we can rely on those to be a lookup key in search jobs results. This job arn is returned and used.

*Testing done:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/amazon-braket/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/integ_tests/conftest.py [hunk 1]
```diff
@@ -162,15 +162,23 @@ def job_failed_name(request):
...
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_completed_name}")
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
+        if job["jobName"] == job_completed_name
+    ][0]
+
+    job = AwsQuantumJob(arn=job_arn)
     return job
```
Comment (speller26) [inline test/integ_tests/conftest.py:172]:
```suggestion
    return AwsQuantumJob(arn=job_arn)
```

File: test/integ_tests/conftest.py [hunk 2]
```diff
@@ -162,15 +162,23 @@ def job_failed_name(request):
...
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_failed_name}")
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
+        if job["jobName"] == job_failed_name
+    ][0]
+
+    job = AwsQuantumJob(arn=job_arn)
     return job
```
Comment (speller26) [inline test/integ_tests/conftest.py:184]:
```suggestion
    return AwsQuantumJob(arn=job_arn)
```

File: test/integ_tests/conftest.py [hunk 3]
```diff
@@ -162,15 +162,23 @@ def job_failed_name(request):
...
 @pytest.fixture(scope="session", autouse=True)
 def completed_quantum_job(aws_session, job_completed_name):
-    account = boto3.client("sts").get_caller_identity().get("Account")
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_completed_name}")
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
+        if job["jobName"] == job_completed_name
+    ][0]
```
Comment (speller26) [inline test/integ_tests/conftest.py:169]: Why not put job name in filter instead of `if job["jobName"] == job_completed_name`?
```suggestion
    job_arn = boto3.client("braket").search_jobs(
        filters=[{"name": job_completed_name}]
    )["jobs"][0]["jobArn"]
```

File: test/integ_tests/conftest.py [hunk 4]
```diff
@@ -162,15 +162,23 @@ def job_failed_name(request):
...
+    job = AwsQuantumJob(arn=job_arn)
     return job
 
 
 @pytest.fixture(scope="session", autouse=True)
 def failed_quantum_job(aws_session, job_failed_name):
-    account = boto3.client("sts").get_caller_identity().get("Account")
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_failed_name}")
+    job_arn = [
```
Comment (speller26) [inline test/integ_tests/conftest.py:177]: ditto

**Review by speller26 — COMMENTED** [review]
Commit prefix should be `test: `

File: test/integ_tests/conftest.py [hunk 5]
```diff
@@ -162,15 +162,21 @@ def job_failed_name(request):
 
 @pytest.fixture(scope="session", autouse=True)
 def completed_quantum_job(aws_session, job_completed_name):
```
Comment (speller26) [inline test/integ_tests/conftest.py:164]: `aws_session` no longer needed
```suggestion
def completed_quantum_job(job_completed_name):
```

File: test/integ_tests/conftest.py [hunk 6]
```diff
@@ -162,15 +162,21 @@ def job_failed_name(request):
...
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
+        if job["jobName"] == job_completed_name
+    ][0]
+
+    return AwsQuantumJob(arn=job_arn)
 
 
 @pytest.fixture(scope="session", autouse=True)
 def failed_quantum_job(aws_session, job_failed_name):
```
Comment (speller26) [inline test/integ_tests/conftest.py:175]: Ditto
```suggestion
def failed_quantum_job(job_failed_name):
```

File: test/integ_tests/conftest.py [hunk 7]
```diff
@@ -161,16 +161,22 @@ def job_failed_name(request):
...
-    account = boto3.client("sts").get_caller_identity().get("Account")
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_completed_name}")
-    return job
+def completed_quantum_job(job_completed_name):
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
+        if job["jobName"] == job_completed_name
+    ][0]
```
Comment (speller26) [inline test/integ_tests/conftest.py:169]: This can get long as more jobs are run; should we filter creation time to start, say, 1 hour ago?

File: test/integ_tests/conftest.py [hunk 8]
```diff
@@ -161,16 +161,22 @@ def job_failed_name(request):
...
 @pytest.fixture(scope="session", autouse=True)
-def completed_quantum_job(aws_session, job_completed_name):
-    account = boto3.client("sts").get_caller_identity().get("Account")
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_completed_name}")
-    return job
+def completed_quantum_job(job_completed_name):
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
```
Comment (speller26) [inline test/integ_tests/conftest.py:167]: This can get long as more jobs are run; should we filter creation time to start, say, 1 hour ago?
```suggestion
        for job in boto3.client("braket").search_jobs(
            filters=[
                {
                    "name": "createdAt",
                    "operator": "GT",
                    "values": [(datetime.now(timezone.utc) - timedelta(hours=1)).isoformat()],
                }
            ]
        )["jobs"]
```

File: test/integ_tests/conftest.py [hunk 9]
```diff
@@ -161,16 +161,22 @@ def job_failed_name(request):
...
 @pytest.fixture(scope="session", autouse=True)
-def failed_quantum_job(aws_session, job_failed_name):
-    account = boto3.client("sts").get_caller_identity().get("Account")
-    region = aws_session.region
-    job = AwsQuantumJob(arn=f"arn:aws:braket:{region}:{account}:job/{job_failed_name}")
-    return job
+def failed_quantum_job(job_failed_name):
+    job_arn = [
+        job["jobArn"]
+        for job in boto3.client("braket").search_jobs(filters=[])["jobs"]
```
Comment (speller26) [inline test/integ_tests/conftest.py:178]: Ditto

File: test/integ_tests/conftest.py (hunk 8 above)
Comment (speller26) [reply test/integ_tests/conftest.py:167]: I guess size isn't the real concern because we're getting only 10 results per page; there _is_ the possibility of the job not being found due to concurrent integ test runs, but that's an extreme edge case.

**Review by speller26 — APPROVED** [review]
